File: app/components/folio/console/tiptap/simple_form_wrap/locale_switch_component.js
window.Folio.Stimulus.register('f-c-tiptap-simple-form-wrap-locale-switch', class extends window.Stimulus.Controller {
  static targets = ['button']

  onAttributeClick (e) {
    const button = e.currentTarget
    const attributeName = button.dataset.attributeName

    // Update active state on buttons
    this.buttonTargets.forEach(btn => {
      btn.classList.toggle('f-c-tiptap-simple-form-wrap-locale-switch__btn--active', btn === button)
    })

    // Dispatch event to parent SimpleFormWrap component
    this.element.dispatchEvent(new CustomEvent('f-c-tiptap-simple-form-wrap-locale-switch:attributeChanged', {
      detail: {
        attributeName
      },
      bubbles: true
    }))
  }
})

File: app/components/folio/console/tiptap/simple_form_wrap/locale_switch_component.rb
# frozen_string_literal: true

class Folio::Console::Tiptap::SimpleFormWrap::LocaleSwitchComponent < Folio::Console::ApplicationComponent
  def initialize(attribute_names:, locales:)
    @attribute_names = attribute_names
    @locales = locales
  end

  def controller_data
    stimulus_controller("f-c-tiptap-simple-form-wrap-locale-switch")
  end

  def attribute_names_with_locales
    @attribute_names.zip(@locales).map do |attribute_name, locale|
      { attribute_name: attribute_name, locale: locale }
    end
  end
end
